design-system/scripts/generate-design-tokens.js

```diff
@@ -21,14 +21,99 @@ const endProgram = (message) => {
   process.exit(1);
 };
 
-const TOKEN_REGEX =
```

Comment:
We were using this regex to parse the tokens names and to check its syntax.

Since I think we need to support more combinations I replaced this regex with a couple of functions which parse and check syntax independently.